### ISSUE TITLE
Generate API documentation For C based projects using DocFx

### ISSUE THREAD

**Issue by nawabahmadreshi** [dotnet/docfx#3656]
- Is it possible to generate API documentation of C projects using DocFx. 
- I have seen certain related issues but no clear answer. 
For reference: 

    - https://github.com/dotnet/docfx/issues/157
    - https://github.com/dotnet/docfx/issues/2484
@vicancy  referring to your comment also: https://github.com/dotnet/docfx/issues/157#issuecomment-300077437

**Comment by superyyrrzz** [comment]
No, DocFX doesn't support for now, and no existing plugin is available for this as far as I know.

**Comment by nawabahmadreshi** [comment]
@superyyrrzz thanks, so we need to write a plugin for the same

**Comment by nawab-samsung** [comment]
@superyyrrzz What is the possibility of doing such a task?

**Comment by superyyrrzz** [comment]
I am sorry that this will not be teams' focus in the near future. You can try code2yaml, while it must need some customization to make it work. (https://github.com/dotnet/docfx/issues/2484#issuecomment-370634586)

**Comment by nawabahmadreshi** [comment]
@superyyrrzz 
I tweaked the code2yaml for C, but how can I use the generated.Yaml file in DocFx.
Kindly guide for the process.

**Comment by superyyrrzz** [comment]
By looking at the `docfx.json` generated by `docfx init`, you can see 2 sections: `metadata` and `build`. `code2yaml` has finished the equivalent `metadata` step. You can copy all the generated YAML files as input, and directly run `docfx build`.

**Comment by nawabahmadreshi** [comment]
@superyyrrzz , the process is that Doxygen runs to parse the files and then an intermediate.XML file is created and that file gets converted to.YAML

I need to know where can I can I change Doxygen configuration file?
Where can I find the .XML file?

**Comment by skanibani** [comment]
> By looking at the `docfx.json` generated by `docfx init`, you can see 2 sections: `metadata` and `build`. `code2yaml` has finished the equivalent `metadata` step. You can copy all the generated YAML files as input, and directly run `docfx build`.

This was a really helpfull comment thanks! :D

**Comment by superyyrrzz** [comment]
@nawabahmadreshi code2yaml tool handles all these. You could get a rough idea by looking into it, or can also refer to doxygen documentation.
